### ISSUE TITLE
Icon Task List applet - enable "Restrict to Workspace" also for "Favorited" apps

### ISSUE THREAD

**Issue by SimoB** [solus-project/budgie-desktop#1803]
Hi, new Budgie (Solus 4) user here, I've enabled the option "Restrict to Workspace" in the Icon Task List applet but seems that is working only for "not pinned" (favorited) applications, so if I have e.g. two instances of firefox (pinned as favorited), one on each workspace, if I click on the task icon (or scroll) it switch between workspaces, that's quite annoying!

Also I think may be related to windows grouping, would it be better if group is restricted only to windows on the same workspace?

**Comment by v-marinkov** [comment]
Yes, yes, yes please! I just tried Solus and was utterly impressed by it.  I do like how scrolling down on a launcher icon minimizes a window, for example. 
Yet, I was disappointed that the Icon Task list has the aforementioned drawback (as does Plank btw). In this respect, Dash to Dock on Gnome is superior in its configurability. This feature is absolutely essential for power users spanning multiple monitors and workspaces.

**Comment by Downsite** [comment]
I also ran into this issue. It is a minor bug, but hinders me quite often. Perhaps #2159 will fix this?
